src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1305,11 +1396,59 @@ bool isAllowedInitOnlySet(BoundExpression receiver)
             return false;
         }
 
-        internal static uint GetInterpolatedStringHandlerConversionEscapeScope(
+        internal uint GetInterpolatedStringHandlerConversionEscapeScope(
             InterpolatedStringHandlerData data,
             uint scopeOfTheContainingExpression)
             => GetValEscape(data.Construction, scopeOfTheContainingExpression);
 
+#nullable enable
+
+        private static bool HasRefStructReturnValue(Symbol symbol)
+        {
+            switch (symbol)
+            {
+                case MethodSymbol method:
+                    return method.MethodKind == MethodKind.Constructor ?
+                        method.ContainingType.IsRefLikeType :
+                        method.ReturnType.IsRefLikeType;
+                case PropertySymbol property:
+                    // PROTOTYPE: Test this case.
+                    return property.ContainingType.IsRefLikeType;
+                default:
+                    return false;
+            }
+        }
+
+        private bool? UseRefEscapeOfInvocationArgument(Symbol symbol, RefKind effectiveRefKind, bool isRefEscape, DeclarationScope scope)
```

Comment:
>UseRefEscapeOfInvocationArgument [](http://example.com/codeflow?start=22&length=32)

I am having a hard time understanding the purpose of this method and mapping its logic to specification. If I understand the spec correctly, an argument can have two contributions:
- can contribute to ref-safe-to-escape
- can contribute to safe-to-escape

Would it be clearer to have two distinct methods answering if an argument should contribute to specific escape mode?

Reply:
> Would it be clearer to have two distinct methods answering if an argument should contribute to specific escape mode?

Perhaps, although an argument will only contribute ref-safe-to-escape or safe-to-escape or neither, but not both, so having two distinct methods may make the caller more complicated if the caller should handle both.